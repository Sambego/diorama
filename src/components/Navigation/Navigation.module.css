.navigation {
  position: fixed;
  bottom: 2rem;
  right: 4rem;
  z-index: 2;
  background: none;
}

.list {
  display: flex;
  margin: 0;
  padding: 0;
  background: none;
}

.item {
  max-width: 2rem;
  margin: 0 2rem;
  padding: 0;
  list-style: none;
}

.previous,
.next {
  transition: border-color 0.2s ease-in-out;
  background: none;
  font-size: 0;
  cursor: pointer;
}

.previous {
  width: 0;
  height: 0;
  border-top: 1.5rem solid transparent;
  border-bottom: 1.5rem solid transparent;
  border-right: 2rem solid var(--color-gray-0);
  border-left: none;

  &:hover {
    border-right: 2rem solid var(--color-gray-2);
  }
}

.next {
  width: 0;
  height: 0;
  border-top: 1.5rem solid transparent;
  border-bottom: 1.5rem solid transparent;
  border-left: 2rem solid var(--color-gray-0);
  border-right: none;

  &:hover {
    border-left: 2rem solid var(--color-gray-2);
  }
}
