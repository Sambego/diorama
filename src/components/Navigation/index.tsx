import React from "react";
import PropTypes from "prop-types";

import styles from "./Navigation.module.css";

interface NavigationProps {
	onPreviousSlide: () => void;
	onNextSlide: () => void;
	className?: string;
	style?: React.CSSProperties;
}
function Navigation({
	onPreviousSlide,
	onNextSlide,
	className,
	style,
}: NavigationProps) {
	return (
		<nav
			className={`${styles.navigation} diorama-navigation ${className}`}
			style={style}
		>
			<ul className={styles.list}>
				<li className={styles.item}>
					<button
						onClick={onPreviousSlide}
						className={`${styles.previous} diorama-previous`}
						type="button"
					>
						Previous
					</button>
				</li>
				<li className={styles.item}>
					<button
						onClick={onNextSlide}
						className={`${styles.next} diorama-next`}
						type="button"
					>
						Next
					</button>
				</li>
			</ul>
		</nav>
	);
}

Navigation.propTypes = {
	className: PropTypes.string,
	onPreviousSlide: PropTypes.func.isRequired,
	onNextSlide: PropTypes.func.isRequired,
	style: PropTypes.shape({}),
};

Navigation.defaultProps = {
	className: "",
	style: {},
};

export default Navigation;
