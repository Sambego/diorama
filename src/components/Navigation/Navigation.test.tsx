import React from "react";
import renderer from "react-test-renderer";
import Navigation from ".";

it("renders correctly", () => {
	const tree = renderer
		.create(
			<Navigation onPreviousSlide={() => true} onNextSlide={() => true} />
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
