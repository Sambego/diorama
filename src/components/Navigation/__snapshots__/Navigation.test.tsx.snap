// Vitest Snapshot v1

exports[`renders correctly 1`] = `
<nav
  className="_navigation_8avy0_1 diorama-navigation "
  style={{}}
>
  <ul
    className="_list_8avy0_9"
  >
    <li
      className="_item_8avy0_16"
    >
      <button
        className="_previous_8avy0_23 diorama-previous"
        onClick={[Function]}
        type="button"
      >
        Previous
      </button>
    </li>
    <li
      className="_item_8avy0_16"
    >
      <button
        className="_next_8avy0_24 diorama-next"
        onClick={[Function]}
        type="button"
      >
        Next
      </button>
    </li>
  </ul>
</nav>
`;
